test/ui/QuerySummaryTest.ts

```diff
@@ -156,5 +202,121 @@ export function QuerySummaryTest() {
+            enableNoResultsFoundMessage: true,
+            noResultsFoundMessage: 'customMessage'
+          });
+
+          test.env.queryStateModel.get = () => 'querySearched';
+          Simulate.query(test.env, { results: FakeResults.createFakeResults(0) });
+
+          const customMessageElement = $$(test.cmp.element).find('.coveo-query-summary-no-results-string');
+
+          expect(customMessageElement.textContent).toBe('customMessage');
+        });
+
+        it(`when there is one query tag in the noResultsFoundMessage
+            it should replace the query tag by the query searched`, () => {
+          test = Mock.optionsComponentSetup<QuerySummary, IQuerySummaryOptions>(QuerySummary, {
+            enableNoResultsFoundMessage: true,
+            noResultsFoundMessage: `customMessage ${queryTag}`
+          });
+
+          test.env.queryStateModel.get = () => 'querySearched';
+          Simulate.query(test.env, { results: FakeResults.createFakeResults(0) });
+
+          const customMessageElement = $$(test.cmp.element).find('.coveo-query-summary-no-results-string');
+
+          expect(customMessageElement.textContent).toBe('customMessage querySearched');
+        });
+
+        it(`when there is multiple query tags in the noResultsFoundMessage
+            it should replace all the query tags by the query searched`, () => {
+          test = Mock.optionsComponentSetup<QuerySummary, IQuerySummaryOptions>(QuerySummary, {
+            enableNoResultsFoundMessage: true,
+            noResultsFoundMessage: `custom ${queryTag} Message ${queryTag}`
+          });
+
+          test.env.queryStateModel.get = () => 'querySearched';
+          Simulate.query(test.env, { results: FakeResults.createFakeResults(0) });
+
+          const customMessageElement = $$(test.cmp.element).find('.coveo-query-summary-no-results-string');
+
+          expect(customMessageElement.textContent).toBe('custom querySearched Message querySearched');
+        });
+      });
+    });
+
+    describe('when a custom no results page is added inside the QuerySummary component', () => {
+      it(`when there are results
+          it should not display the custom no results page`, () => {
+        test.cmp.element.innerHTML = `<div class="${noResultsCssClass}">Custom No Results Page</div>`;
+        Simulate.query(test.env, { results: FakeResults.createFakeResults(10) });
+
+        expect($$(test.cmp.element).find('.coveo-no-results-page-hidden')).not.toBeNull();
+      });
+
+      it(`when there are no results
+          it should display the custom no results page`, () => {
+        test.cmp.element.innerHTML = `<div class="${noResultsCssClass}">Custom No Results Page</div>`;
+
+        Simulate.query(test.env, { results: FakeResults.createFakeResults(0) });
+
+        expect($$(test.cmp.element).find('.coveo-no-results-page-hidden')).toBeNull();
+      });
+
+      it(`when a query tag is added in the custom no results page
```

Comment:
I'd add a test that configures "complex" HTML inside the no results element.

For example, a link which uses the ${queryTag} for it's text or for it's href.

Also, the other edge case I described earlier (eg: the query is empty, the query is ${query}, etc.)

Reply:
- Good point! I added some logic to have the expected behaviour. See the functions `parseQueryTagsInContainer()` and `parseQueryTagsInTextNode()`.

- 👍